### PR TITLE
Quote block: button for cite add/remove

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

All blocks with caption currently have a button to add/remove the caption. But not quote, which has caption-like logic. Let's do the same for caption and reuse the existing caption util.

Not sure if this helps with #15308.

### CODE REVIEW

**Comment by jasmussen** [comment]
I like this, I think we can move forward. Here's a quick GIF, I'll describe it in the next paragraph:

pr test

* Instead of the citation field appearing when selecting the Quote container block, there's a button in the toolbar.
* Click the button to add the citation field, and set focus there, this is the same behavior as how you add a caption to an image.

The GIF also shows two other issues. First off, that even when moving the citation to the toolbar, there's still a layout shift happening when you select the quote block. This appears to be the appender appearing and disappearing. But this issue is not present when selecting a group block where the appender also appears, so this is a layout shift we should fix in this branch.

Secondly, it shows an issue, almost certainly unrelated to this branch, where if you quickly toggle on and off in succession captions on an image, the inline toolbar appars to shift places. Not sure what that is, and separate.

Another thought, let's try the verse icon for the citation button: 
verse icon

If we can add that icon, and fix the layout shift, I'd say we can ship this.

**Comment by ellatrix** [comment]
> so this is a layout shift we should fix in this branch

Not sure why this would be an issue with this branch. Probably a styling issue with the inserter on quote block which seems out of scope for this PR. I don't see the current theme I have enabled.

Updating the icon now.

**Comment by ellatrix** [comment]
Icon changed.

File: packages/block-library/src/pullquote/edit.js [hunk 1]
```diff
@@ -66,32 +66,26 @@ function PullQuoteEdit( {
...
-							__unstableMobileNoFocusOnMount
-							textAlign="center"
-							__unstableOnSplitAtEnd={ () =>
-								insertBlocksAfter(
-									createBlock( getDefaultBlockName() )
-								)
-							}
-						/>
-					) }
+					<Caption
```
Comment (mtias) [inline packages/block-library/src/pullquote/edit.js]: Should we skip doing this for the pullquote block?

**Review by mtias — APPROVED** [review]
Thanks for looking at this. This is feature that doesn't warrant being so visible, and it's nice to address that long standing problem of the hanging placeholder.

**Comment by jasmussen** [comment]
> Not sure why this would be an issue with this branch. Probably a styling issue with the inserter on quote block which seems out of scope for this PR. I don't see the current theme I have enabled.

Seems fine to fix separately, but the the issue is invisible in trunk, but super visible here, so if we merge this one, we will need to fix the other issue shortly after.

**Comment by ellatrix** [comment]
@jasmussen What if we just remove the appender? It doesn't seem so useful in the quote block. You can press enter and use the empty inserter.

**Comment by jasmussen** [comment]
> What if we just remove the appender? It doesn't seem so useful in the quote block. You can press enter and use the empty inserter.

IMO we need to track down why the quote block uses a different appender than the group block. It's outputting different CSS classes. As a container block of anything, it seems prudent to be consistent with other containers.

**Comment by dcalhoun** [comment]
👋🏻 @ellatrix. I noticed the mobile unit tests fail currently. It appears the https://github.com/WordPress/gutenberg/pull/59097 changes that were merged into this branch are no longer present. Was the removal an oversight during a force push?

**Comment by ellatrix** [comment]
@dcalhoun Ah, thanks! I missed that. I re-applied your change but without the pull quote changes which were reverted for the web side as well.

**Pull request merged**